### PR TITLE
Animation: port to the new module definition syntax (stage-2 variant)

### PR DESCRIPTION
**Pull request by ChALkeR**
This ports `Animation` to the new module definition syntax to showcase my proposal in #260.

This is alternative to #261 and uses an even nicer syntax, but requires a ~~`stage-1`~~ `stage-2` transform to support [ES Class Fields & Static Properties](https://github.com/jeffmo/es-class-fields-and-static-properties) ~~`stage-1`~~ `stage-2` proposal. **Upd:** it's `stage-2` now.

`stage-2` means «The committee expects the feature to be developed and eventually included in the standard». See [TC39 Process](https://tc39.github.io/process-document/) for more info on that.

~~I doubt that we want this atm, perhaps we should wait until it reaches `stage-2` or `stage-3`.~~
Not sure if we want to adopt this on the current stage or if we should wait for it to reach `stage-3`, which could take some time.

I landed runtime support for this syntax in 188e9fd (i.e. for passing a `function` to `registerQmlType`).

/cc @akreuzkamp @Plaristote

### CODE REVIEW

**Comment by ChALkeR** [comment]
Status update: [ES Class Fields & Static Properties](https://github.com/jeffmo/es-class-fields-and-static-properties)  is now at `stage-2` aka `Draft` (see https://tc39.github.io/process-document/).

It has now has a formal specification at https://tc39.github.io/proposal-class-public-fields/, and is more likely to be accepted as a future ECMAScript language feature.

We could reconsider this. I updated the PR. @akreuzkamp, @Plaristote, what do you think?

**Comment by ChALkeR** [comment]
Tests broken because of coverage being broken, I will take a look at that if we choose to adopt this syntax.

File: src/modules/QtQuick/Animation.js [hunk 1]
```diff
@@ -1,19 +1,21 @@
-QmlWeb.registerQmlType({
-  module: "QtQuick",
-  name: "Animation",
-  versions: /.*/,
-  baseClass: "QtQml.QtObject",
-  enums: {
+QmlWeb.registerQmlType(class {
+  static module = "QtQuick";
+  static element = "Animation";
```
Comment (akreuzkamp) [inline src/modules/QtQuick/Animation.js]: why did you rename that property to element? I think "element" is something you shouldn't use as a name for (almost) anything, because it's too vague... What's wrong about `name`?

**Comment by akreuzkamp** [comment]
does it really work without any changes in `registerQmlType`?

In any way I do like this syntax :)

**Comment by ChALkeR** [comment]
@akreuzkamp The changes in `registerQmlType` landed in 188e9fd, it currently supports all three syntaxes: `options, constructor`, a single constructor with static options, a single options object with a `constructor` property. The last one is old and unused, but it's still there because it's used for internal representation.

File: src/modules/QtQuick/Animation.js (hunk 1 above)
Comment (ChALkeR) [inline src/modules/QtQuick/Animation.js]: `name` is already present for all classes (and functions), it's a string there, and in this specific case it's an empty string because the class is unnamed.

```shell
> (function x(){}).name
'x'
> (class {}).name
''
> (class Foo {}).name
'Foo'
```

Yes, `element` isn't perfect, but I wasn't able to come up with something better. Any suggestions?

Comment (ChALkeR) [inline src/modules/QtQuick/Animation.js]: Otoh, we could just write `class Animation {` and keep `.name`. What do you think?

Comment (ChALkeR) [inline src/modules/QtQuick/Animation.js]: @akreuzkamp We could do both — use `.name` by default and write `class Animation {` in our code, but allow users to override it with a static property so that they could use anonymous functions there. How should that property be named then?

**Comment by ChALkeR** [comment]
Ok, I'm going to postpone this.

**Comment by ChALkeR** [comment]
Ok, this passes the tests now, and coverage works.
